.github/workflows/e2e-test.yaml

```diff
@@ -95,3 +99,113 @@ jobs:
           command: |
             export REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
             python3 e2etest/reconnection/main.py
+  e2e-test-grpc:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11"]
+    steps:
+      - name: Checkout astarte-device-sdk-python
+        uses: actions/checkout@v4
+        with:
+          path: astarte-device-sdk-python
+      - name: Checkout astarte-message-hub
+        uses: actions/checkout@v4
+        with:
+          repository: astarte-platform/astarte-message-hub
+          path: astarte-message-hub
+          ref: v0.5.2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Protoc
+        run: |
+          curl -LO $PB_REL/download/v24.3/protoc-24.3-linux-x86_64.zip
+          unzip protoc-24.3-linux-x86_64.zip -d $HOME/.local
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - name: Create Astarte Cluster
+        id: astarte
+        uses: astarte-platform/astarte-cluster-action@v1
+        with:
+          astarte_version: "1.1"
+      - name: Install interface
+        run: |
+          astartectl realm-management interfaces sync $GITHUB_WORKSPACE/astarte-device-sdk-python/e2etest/interfaces/*.json --non-interactive
+          astartectl realm-management interfaces ls
+      - name: Setup env variables common for all devices
+        run: |
+          echo "E2E_REALM=test" >> $GITHUB_ENV
+          APPENGINE_TOKEN=$(astartectl utils gen-jwt appengine)
+          echo "E2E_APPENGINE_TOKEN=$APPENGINE_TOKEN" >> $GITHUB_ENV
+          echo "E2E_APPENGINE_URL=https://api.autotest.astarte-platform.org/appengine" >> $GITHUB_ENV
+          echo "E2E_PAIRING_URL=https://api.autotest.astarte-platform.org/pairing" >> $GITHUB_ENV
+      - name: Setup env variables for device 1
+        run: |
+          DEVICE_1_ID=$(astartectl utils device-id generate-random)
+          echo "E2E_DEVICE_1_ID=$DEVICE_1_ID" >> $GITHUB_ENV
+          echo "E2E_DEVICE_3_ID=$DEVICE_1_ID" >> $GITHUB_ENV
```

Comment:
Why do you use two different (1,3) device_id?

Reply:
Ah, this is quite an ugly workaround to have the same tests used for the mqtt transport work for grpc.
Right now each test has a device number hardcoded (base is 1, persistency is 2 and reconnection is 3).
The device ID is necessary also when running the tests for grpc as HTTP transactions are used to verify the device behavior.
With grpc there is a single device, so the required environment variables have all the same value.

Reply:
I'll fix this by passing the device number as a command line argument.